pkg/branchcuts/bumper/release-controller-config-bumper.go

```diff
@@ -0,0 +1,212 @@
+
+	if err := bumpReleaseCheck(releaseConfig, b.mm.Major); err != nil {
+		return releaseConfig, err
+	}
+	if err := bumpReleasePublish(releaseConfig, b.mm.Major); err != nil {
+		return releaseConfig, err
+	}
+	if err := bumpReleaseVerification(releaseConfig, b.mm.Major); err != nil {
+		return releaseConfig, err
+	}
+
+	setMapIfNil(&releaseConfig.Check, map[string]ReleaseCheck{})
+	setMapIfNil(&releaseConfig.Verify, map[string]ReleaseVerification{})
+	setMapIfNil(&releaseConfig.Publish, map[string]ReleasePublish{})
+
+	return releaseConfig, nil
+}
+
+func bumpReleaseCheck(releaseConfig *ReleaseConfig, currentMajor int) error {
+	if releaseConfig.Check == nil {
+		return nil
+	}
+	check := make(map[string]ReleaseCheck)
+	for k, v := range releaseConfig.Check {
+		if v.ConsistentImages != nil {
+			if err := ReplaceWithNextVersionInPlace(&v.ConsistentImages.Parent, currentMajor); err != nil {
+				return err
+			}
+		}
+		check[k] = v
+	}
+	releaseConfig.Check = check
+	return nil
+}
+
+func bumpReleasePublish(releaseConfig *ReleaseConfig, currentMajor int) error {
+	if releaseConfig.Publish == nil {
+		return nil
+	}
+	publish := make(map[string]ReleasePublish)
+	for k, v := range releaseConfig.Publish {
+		if v.VerifyBugs != nil && v.VerifyBugs.PreviousReleaseTag != nil {
+			if err := ReplaceWithNextVersionInPlace(&v.VerifyBugs.PreviousReleaseTag.Name, currentMajor); err != nil {
+				return err
+			}
+			if err := ReplaceWithNextVersionInPlace(&v.VerifyBugs.PreviousReleaseTag.Tag, currentMajor); err != nil {
+				return err
+			}
+		}
+		if v.ImageStreamRef != nil {
+			if err := ReplaceWithNextVersionInPlace(&v.ImageStreamRef.Name, currentMajor); err != nil {
+				return err
+			}
+		}
+		if v.TagRef != nil {
+			if err := ReplaceWithNextVersionInPlace(&v.TagRef.Name, currentMajor); err != nil {
+				return err
+			}
+		}
+		publish[k] = v
+	}
+	releaseConfig.Publish = publish
+	return nil
+}
+
+func bumpReleaseVerification(releaseConfig *ReleaseConfig, currentMajor int) error {
+	if releaseConfig.Verify == nil {
+		return nil
+	}
+	verify := make(map[string]ReleaseVerification)
+	for k, v := range releaseConfig.Verify {
+		if v.ProwJob != nil {
+			if err := ReplaceWithNextVersionInPlace(&v.ProwJob.Name, currentMajor); err != nil {
+				return err
+			}
+		}
+		newKey, err := ReplaceWithNextVersion(k, currentMajor)
+		if err != nil {
+			return err
+		}
+		verify[newKey] = v
+	}
+	releaseConfig.Verify = verify
+	return nil
+}
+
+func (b *ReleaseControllerConfigBumper) Marshall(releaseConfig *ReleaseConfig,
+	bumpedFilename, dir string) error {
+	absolutePath := path.Join(dir, bumpedFilename)
+	raw, err := json.MarshalIndent(releaseConfig, "", jsonIndent)
+	if err != nil {
+		return fmt.Errorf("failed to marshall release config")
+	}
+	if err := ioutil.WriteFile(absolutePath, raw, 0666); err != nil {
+		return fmt.Errorf("failed to write file %s", absolutePath)
+	}
+	return nil
+}
+
+func setMapIfNil[T any](x *map[string]T, y map[string]T) {
```

Comment:
I have factored this code out:
```golang
// m is a map of type map[string]T, where T is any
if m == nil {
  m = make(map[string]T)
}
```
that would be have otherwise replicated at least three times.
Happy to change it if you know a more idiomatic approach.

Reply:
In the place you are defining the map you should create it instead. Then there will be no need to do this check. Also, you can just do the `if nil then make it` as well.